alicloud/resource_alicloud_alikafka_instance.go

```diff
@@ -0,0 +1,332 @@
+package alicloud
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/alikafka"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudAlikafkaInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudAlikafkaInstanceCreate,
+		Read:   resourceAlicloudAlikafkaInstanceRead,
+		Update: resourceAlicloudAlikafkaInstanceUpdate,
+		Delete: resourceAlicloudAlikafkaInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"topic_quota": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"disk_type": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"disk_size": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"deploy_type": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"io_max": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"eip_max": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vswitch_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
```

Comment:
I think the `vpc_id` and `zone_id` can be removed and them can be get by `vswitch_id`.